lib/libzfs/libzfs_sendrecv.c

```diff
@@ -3929,7 +3929,7 @@ zfs_receive_one(libzfs_handle_t *hdl, int infd, const char *tosnap,
 				zfs_error_aux(hdl, dgettext(TEXT_DOMAIN,
 				    "destination has snapshots (eg. %s)\n"
 				    "must destroy them to overwrite it"),
-				    name);
+				    zc.zc_name);
```

Comment:
I must be missing something but given the `strcpy()` on line 3909 above, does this actually change the output?

Reply:
`ioctl(ZFS_IOC_SNAPSHOT_LIST_NEXT)` will update `zc.zc_name`:

```shell
root@linux:/usr/src/zfs# git diff
diff --git a/lib/libzfs/libzfs_sendrecv.c b/lib/libzfs/libzfs_sendrecv.c
index 4a620a9..34ddd6a 100644
--- a/lib/libzfs/libzfs_sendrecv.c
+++ b/lib/libzfs/libzfs_sendrecv.c
@@ -3929,7 +3929,7 @@ zfs_receive_one(libzfs_handle_t *hdl, int infd, const char *tosnap,
                                zfs_error_aux(hdl, dgettext(TEXT_DOMAIN,
                                    "destination has snapshots (eg. %s)\n"
                                    "must destroy them to overwrite it"),
-                                   name);
+                                   zc.zc_name);
                                err = zfs_error(hdl, EZFS_EXISTS, errbuf);
                                goto out;
                        }
root@linux:/usr/src/zfs# zfs send $POOLNAME/fs@snap1 | zfs recv -F $POOLNAME/recvfs
cannot receive new filesystem stream: destination has snapshots (eg. pp/recvfs@snap-orig)
must destroy them to overwrite it
root@linux:/usr/src/zfs# zfs send $POOLNAME/fs@snap1 > /var/tmp/send
root@linux:/usr/src/zfs# gdb -q zfs -ex "b lib/libzfs/libzfs_sendrecv.c:3910" -ex "run recv -F $POOLNAME/recvfs < /var/tmp/send"
Reading symbols from zfs...done.
No source file named lib/libzfs/libzfs_sendrecv.c.
Make breakpoint pending on future shared library load? (y or [n]) y
Breakpoint 1 (lib/libzfs/libzfs_sendrecv.c:3910) pending.
Starting program: /usr/sbin/zfs recv -F pp/recvfs < /var/tmp/send
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".

Breakpoint 1, zfs_receive_one (hdl=0x7fffffffa0f0, infd=-36624, tosnap=0x7fffffffa0f0 "pp/recvfs", originsnap=0x7fffffff70f0 "pp/recvfs", flags=0x7fffffffea30, drr=0x7fffffffdce0, drr_noswap=0x7fffffffde20, 
    sendfs=0x7fffffffe4a0 "pp/fs", stream_nv=0x0, stream_avl=0x0, top_zfs=0x7fffffffe930, cleanup_fd=-24336, action_handlep=0x7fffffffe938, finalsnap=0x0, cmdprops=0x621ab0) at libzfs_sendrecv.c:3920
3920				if (!flags->force) {
(gdb) p zc.zc_name
$1 = "pp/recvfs", '\000' <repeats 4086 times>
(gdb) b write
Breakpoint 2 at 0x7ffff6a1dbf0: write. (2 locations)
(gdb) c
Continuing.

Breakpoint 2, write () at ../sysdeps/unix/syscall-template.S:81
81	../sysdeps/unix/syscall-template.S: No such file or directory.
(gdb) bt
#0  write () at ../sysdeps/unix/syscall-template.S:81
#1  0x00007ffff69b6423 in _IO_new_file_write (f=0x7ffff6ce8060 <_IO_2_1_stderr_>, data=0x7fffffff4100, n=124) at fileops.c:1253
#2  0x00007ffff69b5ae3 in new_do_write (fp=fp@entry=0x7ffff6ce8060 <_IO_2_1_stderr_>, 
    data=data@entry=0x7fffffff4100 "cannot receive new filesystem stream: destination has snapshots (eg. pp/recvfs@snap-orig)\nmust destroy them to overwrite it\n", to_do=to_do@entry=124) at fileops.c:530
#3  0x00007ffff69b6a36 in _IO_new_file_xsputn (f=0x7ffff6ce8060 <_IO_2_1_stderr_>, data=<optimized out>, n=124) at fileops.c:1335
#4  0x00007ffff698db9d in buffered_vfprintf (s=s@entry=0x7ffff6ce8060 <_IO_2_1_stderr_>, format=format@entry=0x7ffff77a0297 "%s: %s\n", args=args@entry=0x7fffffff6718) at vfprintf.c:2369
#5  0x00007ffff698837e in _IO_vfprintf_internal (s=0x7ffff6ce8060 <_IO_2_1_stderr_>, format=0x7ffff77a0297 "%s: %s\n", ap=ap@entry=0x7fffffff6718) at vfprintf.c:1296
#6  0x00007ffff6992ce7 in __fprintf (stream=<optimized out>, format=format@entry=0x7ffff77a0297 "%s: %s\n") at fprintf.c:32
#7  0x00007ffff777a632 in zfs_verror (hdl=0x620060, error=2008, fmt=<optimized out>, ap=<optimized out>) at libzfs_util.c:328
#8  0x00007ffff777a952 in zfs_error_fmt (hdl=<optimized out>, error=<optimized out>, fmt=<optimized out>) at libzfs_util.c:349
#9  0x00007ffff7775fa2 in zfs_receive_one (hdl=0x2, infd=-48896, tosnap=0x7fffffffa0f0 "pp/recvfs@snap-orig", originsnap=0x7fffffff70f0 "pp/recvfs", flags=0x620060, drr=0x7fffffffdce0, 
    drr_noswap=0x7fffffffde20, sendfs=0x7fffffffe4a0 "pp/fs", stream_nv=0x0, stream_avl=0x0, top_zfs=0x7fffffffe930, cleanup_fd=-154245472, action_handlep=0x7fffffffe938, finalsnap=0x0, cmdprops=0x621ab0)
    at libzfs_sendrecv.c:3933
#10 0x00007ffff7776a5e in zfs_receive_impl (hdl=0x620060, tosnap=0x7fffffff4100 "cannot receive new filesystem stream: destination has snapshots (eg. pp/recvfs@snap-orig)\nmust destroy them to overwrite it\n", 
    tosnap@entry=0x621b20 "pp/recvfs", originsnap=0x7c <error: Cannot access memory at address 0x7c>, flags=0x1, flags@entry=0x7fffffffea30, infd=-134326400, infd@entry=0, 
    sendfs=0x7ffff698d99a <_IO_vfprintf_internal+22490> "\200\275(\373\377\377", sendfs@entry=0x0, stream_nv=0x0, stream_avl=0x0, top_zfs=0x7fffffffe930, cleanup_fd=6, action_handlep=0x7fffffffe938, 
    finalsnap=0x0, cmdprops=0x621ab0) at libzfs_sendrecv.c:4546
#11 0x00007ffff7778fa5 in zfs_receive (hdl=0x620060, tosnap=0x621b20 "pp/recvfs", props=0x621ab0, flags=0x7fffffffea30, infd=0, stream_avl=0x0) at libzfs_sendrecv.c:4625
#12 0x000000000040de4c in zfs_do_receive (argc=1, argv=0x621d58) at zfs_main.c:4344
#13 0x0000000000405646 in main (argc=4, argv=0x7fffffffecb8) at zfs_main.c:8027
(gdb) frame 9
#9  0x00007ffff7775fa2 in zfs_receive_one (hdl=0x2, infd=-48896, tosnap=0x7fffffffa0f0 "pp/recvfs@snap-orig", originsnap=0x7fffffff70f0 "pp/recvfs", flags=0x620060, drr=0x7fffffffdce0, 
    drr_noswap=0x7fffffffde20, sendfs=0x7fffffffe4a0 "pp/fs", stream_nv=0x0, stream_avl=0x0, top_zfs=0x7fffffffe930, cleanup_fd=-154245472, action_handlep=0x7fffffffe938, finalsnap=0x0, cmdprops=0x621ab0)
    at libzfs_sendrecv.c:3933
3933					err = zfs_error(hdl, EZFS_EXISTS, errbuf);
(gdb) p zc.zc_name
$2 = "pp/recvfs@snap-orig", '\000' <repeats 4076 times>
(gdb) 
```